Rakefile

```diff
@@ -1,6 +1,37 @@
 require 'bundler'
 require 'bundler/gem_tasks'
+require 'bump'
+require 'github_changelog_generator'
 
 task :spec do
   require File.expand_path('spec/run')
 end
+
+desc 'Bump minor version, create changelog, tag, push to github, and publish to rubygems.'
+task :publish do
+  # Bumps the version.rb patch version (there are gems that can handle this for us)
+  Rake::Task[:minor].invoke
+
+  # Runs github_changelog_generator to update the CHANGELOG.rb per the release guide
+  # need to figure out how to check for https://github.com/skywinder/github-changelog-generator#github-token
+  sh 'github_changelog_generator'
+
+  # Commits those as a change to master (git commit -am "Preparing the #{future_version} release" && git push)
+  future_version = Bump::Bump.current
+  sh "git commit -am 'Preparing for #{future_version} release' && git push"
+
+  # Runs git tag to take the release (git tag -a "Releasing #{future_version}" && git push --tags)
+  sh "git tag -a 'Releasing #{future_version}' && git push --tags"
+
+  # need to add rake release here :)
```

Comment:
`rake release` will do the tagging for you, so when that is added you won't need the above line